src/test/java/org/openhab/tools/analysis/checkstyle/test/PackageExportsNameCheckTest.java

```diff
@@ -0,0 +1,105 @@
+/**
+ * Copyright (c) 2010-2017 by the respective copyright holders.
+ *
+ * All rights reserved. This program and the accompanying materials
+ * are made available under the terms of the Eclipse Public License v1.0
+ * which accompanies this distribution, and is available at
+ * http://www.eclipse.org/legal/epl-v10.html
+ */
+package org.openhab.tools.analysis.checkstyle.test;
+
+import static org.openhab.tools.analysis.checkstyle.api.CheckConstants.*;
+
+import java.io.File;
+import java.io.IOException;
+
+import org.junit.BeforeClass;
+import org.junit.Test;
+import org.openhab.tools.analysis.checkstyle.PackageExportsNameCheck;
+import org.openhab.tools.analysis.checkstyle.api.AbstractStaticCheckTest;
+
+import com.puppycrawl.tools.checkstyle.DefaultConfiguration;
+import com.puppycrawl.tools.checkstyle.api.Configuration;
+import com.puppycrawl.tools.checkstyle.utils.CommonUtils;
+
+/**
+ * Test for {@link PackageExportsNameCheck }
+ *
+ * @author Аleksandar Кovachev - Initial contribution
+ * @author Petar Valchev - Added a test for non existent packages
+ *
+ */
+public class PackageExportsNameCheckTest extends AbstractStaticCheckTest {
+    private static final String CORRECT_NAMING_OF_NOT_EXPORTED_PACKAGES_MESSAGE = "The package %s"
+            + " should be marked as \"internal\" if it will not be exported.";

```

Comment:
I think the sentence is not correct:

... should be marked as internal if it is not exported